crates/forge/bin/cmd/coverage.rs

```diff
@@ -93,6 +93,15 @@ impl CoverageArgs {
     fn build(&self, config: &Config) -> Result<(Project, ProjectCompileOutput)> {
         // Set up the project
         let mut project = config.create_project(false, false)?;
+
+        // Set a different artifacts path for coverage. `out/coverage`.
+        // This is done to avoid overwriting the artifacts of the main build that maybe built with
+        // different optimizer settings or --via-ir. Optimizer settings are disabled for
+        // coverage builds.
+        let coverage_artifacts_path = project.artifacts_path().join("coverage");
+        project.paths.artifacts = coverage_artifacts_path.clone();
+        project.paths.build_infos = coverage_artifacts_path.join("build-info");
```

Comment:
cache path also needs to be updated. could we add a test that runs `forge build` -> `forge coverage` -> `forge build` -> `forge coverage` and asserts that 2 latest commands do not compile anything?

Reply:
Test has been added